src/test/java/bio/terra/service/snapshotbuilder/CriteriaQueryBuilderTest.java

```diff
@@ -0,0 +1,281 @@
+import bio.terra.service.snapshotbuilder.query.Query;
+import bio.terra.service.snapshotbuilder.query.TablePointer;
+import bio.terra.service.snapshotbuilder.query.TableVariable;
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag(Unit.TAG)
+public class CriteriaQueryBuilderTest {
+
+  @Test
+  void generateRangeCriteriaFilterProducesCorrectSql() {
+    SnapshotBuilderProgramDataRangeCriteria rangeCriteria = generateRangeCriteria();
+    FilterVariable filterVariable =
+        new CriteriaQueryBuilder().generateFilterForRangeCriteria(rangeCriteria);
+
+    // Table name is null because there is no alias generated until it is rendered as a full query
+    assertThat(
+        "The sql generated is correct",
+        filterVariable.renderSQL(),
+        equalTo("(null.range_column_name >= 0 AND null.range_column_name <= 100)"));
+    assertThat(
+        "The filter just has person as the table",
+        filterVariable.getTables().stream()
+            .map(TableVariable::getTablePointer)
+            .map(TablePointer::tableName)
+            .toList(),
+        equalTo(List.of("person")));
+  }
+
+  @Test
+  void generateListCriteriaFilterProducesCorrectSql() {
+    SnapshotBuilderProgramDataListCriteria listCriteria = generateListCriteria();
+    FilterVariable filterVariable =
+        new CriteriaQueryBuilder().generateFilterForListCriteria(listCriteria);
+
+    // Table name is null because there is no alias generated until it is rendered as a full query
+    assertThat(
+        "The sql generated is correct",
+        filterVariable.renderSQL(),
+        equalTo("null.list_column_name IN (0,1,2)"));
+    // The person table is reachable by traversing through the table variables, but isn't directly
+    // available
+    assertThat(
+        "The filter just has the concept table, but not the person table",
+        filterVariable.getTables().stream()
+            .map(TableVariable::getTablePointer)
+            .map(TablePointer::tableName)
+            .toList(),
+        equalTo(List.of("person")));
+  }
+
+  @Test
+  void generateDomainCriteriaFilterProducesCorrectSql() {
+    SnapshotBuilderDomainCriteria domainCriteria = generateDomainCriteria();
+    FilterVariable filterVariable =
+        new CriteriaQueryBuilder().generateFilterForDomainCriteria(domainCriteria);
+
+    // Table name is null because there is no alias generated until it is rendered as a full query
+    assertThat(
+        "The sql generated is correct",
+        filterVariable.renderSQL(),
+        equalTo(
+            "null.person_id IN (SELECT c.person_id FROM condition_occurrence AS c  JOIN concept_ancestor AS c0 ON c0.ancestor_concept_id = c.condition_concept_id WHERE (c.condition_concept_id = 0 OR c0.ancestor_concept_id = 0))"));
```

Comment:
for sql string comparison you could use Matchers.equalToCompressingWhiteSpace() if you want/have had problems with tests be fragile bc of whitespace/formatting

Reply:
I switched to `equalToCompressingWhiteSpace()`. I think we do want the exactness rather than using containsString.